ballista/rust/scheduler/src/state/shuffle_reaper.rs

```diff
@@ -0,0 +1,52 @@
+use ballista_core::client::BallistaClient;
+use ballista_core::serde::scheduler::PartitionLocation;
+use log::error;
+use std::collections::HashMap;
+use tokio::sync::mpsc::Receiver;
+
+pub struct ShuffleReaper {
+    receiver: Receiver<Vec<PartitionLocation>>,
+}
+
+impl ShuffleReaper {
+    pub fn new(receiver: Receiver<Vec<PartitionLocation>>) -> Self {
+        Self { receiver }
+    }
+
+    pub fn start(self) {
+        tokio::task::spawn(async move {
+            let mut clients: HashMap<String, BallistaClient> = Default::default();
+            let mut receiver = self.receiver;
+            while let Some(locations) = receiver.recv().await {
+                for location in locations {
+                    if let Some(client) = clients.get_mut(&location.executor_meta.id) {
+                        match client.delete_partition(&location.path).await {
```

Comment:
I think such pattern is more readable when written using `unwrap_or_else`.

So something like this
```rust
match something {
  Ok(()) => (),
  Err(e) => something_else
```
becomes
```rust
something.unwrap_or_else(|| something_else)
```